week4/routes/cohorts.js

```diff
@@ -0,0 +1,182 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../db/client") 
+
+router.get('/new', (req, res)=>{
+    res.render('new')
+})
+
+router.post('/', (req, res)=>{
+    const logoUrl = req.body.logoUrl;
+    const name = req.body.name;
+    const members = req.body.members;
+    const newCohort = {
+        logoUrl: logoUrl,
+        name: name,
+        members: members
+    };
+
+    knex 
+        .insert(newCohort)
+        .into('cohorts')
+        .returning("*")
+        .then((cohorts)=>{
+            console.log(cohorts);
+            const [cohort] = cohorts;
+            res.redirect(`/cohorts/${cohort.id}`)
+        })
+})
+
+router.get('/:id/form', (req, res)=>{
+    const id = req.params.id;
+
+    knex  
+        .select("*")
+        .from("cohorts")
+        .where({ id: id })
+        .first() 
+        .then((cohort)=>{ 
+            console.log(cohort);
+            const method = req.query.method;
+            const quantity = req.query.quantity;
+            const members = cohort.members;
+            
+        
+    function teamBuilder (strOfMembers, quantity, method){
```

Comment:
It is unadvised to have this function defined within another function. There are cases where you might want to do this, but in this case it is bad form.

It makes this function massive, and unruly and hard to read/follow. 
I liked how you put it in a separate file earlier, that is the best way to do it.
From there, I would use `module.exports` to export that function from the other file, and then `require` that function here in order to use it.
That would be the best way to do this.